flat_pack_library.scad

```diff
@@ -0,0 +1,170 @@
+*/
+
+
+/*
+##module: intside_cuts_debug
+Create a set of finger-joint cuts that result in two larger cuts taken at the outside
+edge with debugging text
+  ###parameters:
+    *length*      (real)           length of edge
+    *finger*      (real)           length of each individual finger
+    *material*    (real)           thickness of material - sets cut depth
+    *center*      (boolean)        center the set of fingers with respect to origin
+    *font*        (string)         name of font 
+*/
+
+
+module inside_cuts_debug(length=6, finger=1, material=1, center=false, font="Liberation Sans") {  
+
+  x_translation = center==false ? length/2 : 0;
+  y_translation = center==false ? material*1.5 : material; 
+
+  inside_cuts(length=length, finger=finger, material=material, center=center);
+
+  debugText = finger>=length/3 ? "ERR: finger>1/3 length" : "inside cuts";
+
+  translate([x_translation, y_translation,  0])
+  text(text=debugText, size = length*.1, halign = "center", font = font);
+
+
+} // end inside debug
+
+
+/*
+##module: outside_cuts_debug
+Create a set of finger-joint cuts that result in two larger cuts taken at the inside
+edge with debugging text
+  ###parameters:
+    *length*      (real)           length of edge
+    *finger*      (real)           length of each individual finger
+    *material*    (real)           thickness of material - sets cut depth
+    *center*      (boolean)        center the set of fingers with respect to origin
+    *font*        (string)         name of font 
+*/
+
+module outside_cuts_debug(length=6, finger=1, material=1, center=false, font="Liberation Sans") {  
+
+  x_translation = center==false ? length/2 : 0;
+  y_translation = center==false ? material*1.5 : material; 
+
+  outside_cuts(length=length, finger=finger, material=material, center=center);
+
+  debugText = finger>=length/3 ? "ERR: finger>1/3 length" : "outside cuts";
+
+  translate([x_translation, y_translation,  0])
+  text(text=debugText, size = length*.1, halign = "center", font = font);
+
+
+} // end outside debug
+
+/*
+##module: outside_cuts
+Create a set of finger-joint cuts that result in two larger cuts taken at the outside
+edge
+  ###parameters:
+    *length* (real)         length of edge
+    *finger* (real)         length of each individual finger
+    *material* (real)       thickness of material - sets cut depth
+    *center* (boolean)         center the set of fingers with respect to origin
+*/
+
+module outside_cuts(length=6, finger=1, material=1, text=false, center=false, font="Liberation Sans") {
+  // overage to ensure that all cuts are completed
+  overage = 0.0001;
+
+
+  //maximum possible divisions for this length
+  max_divisions = floor(length/finger);
+  //number of usable divisions that fall completely within the edge
+  //for this implentation the number of divisions must be odd
+  usable_divsions = max_divisions%2==0 ? max_divisions-3 : max_divisions-2;
```

Comment:
Whoops, just noticed a typo in `usable_divisions` here (and in the inside cuts module too)